src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4041,6 +4076,32 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
             {
                 analyzedArguments.Free();
             }
+
+            static void validateRecordCopyConstructor(NamedTypeSymbol containingType, MethodSymbol constructor, NamedTypeSymbol baseType, MethodSymbol baseCtor, Location errorLocation, DiagnosticBag diagnostics)
+            {
+                // Are we dealing with a user-defined copy constructor in a record type?
+                if (containingType is SourceNamedTypeSymbol sourceType &&
+                    sourceType.IsRecord() &&
+                    SynthesizedRecordCopyCtor.FindCopyConstructor(containingType).Equals(constructor, TypeCompareKind.ConsiderEverything))
+                {
+                    if (baseType.SpecialType == SpecialType.System_Object)
+                    {
+                        return;
+                    }
+
+                    // Unless the base type is 'object', the constructor should invoke the base type copy constructor
+                    var correctBaseCopyCtor = SynthesizedRecordCopyCtor.FindCopyConstructor(baseType);
+                    if (correctBaseCopyCtor is null)
+                    {
+                        // We'll have reported a problem with the base type of the record elsewhere
```

Comment:
>// We'll have reported a problem with the base type of the record elsewhere [](start = 24, length = 75)

The base can be declared in a different compilation. It looks like we might fail to report any error in this scenario (errors in a different compilation aren't good enough).

Reply:
Thanks, indeed

---
In reply to: [438506165](https://github.com/dotnet/roslyn/pull/45022#discussion_r438506165) [](ancestors = 438506165)